activestorage/lib/active_storage/analyzer/image_analyzer/vips.rb

```diff
@@ -36,6 +36,11 @@ def rotated_image?(image)
         false
       end
 
+      def opaque?(image)
+        return true unless image.has_alpha?
+        image[image.bands - 1].min == 255
```

Comment:
This from [an issue](https://github.com/libvips/libvips/issues/547) where jcupitt, maintainer of vips and ruby-vips, explained how to detect opaqueness.